src/microvm/src/vmm/microvm/mod.rs

```diff
@@ -62,33 +66,53 @@ use ::sys::ipc::{
 
 pub struct Vmm {
     _gateway_tx: Sender<Message>,
-    _io_thread: Option<JoinHandle<Result<()>>>,
+    io_thread: Option<JoinHandle<Result<()>>>,
     _memory_thread: JoinHandle<Result<()>>,
-    microvm: Arc<Mutex<MicroVm>>,
+    vcpu_thread: JoinHandle<Result<u16>>,
+    _microvm: Arc<Mutex<MicroVm>>,
+    snapshot_rx: Receiver<SnapshotCommand>,
+    _paused_tx: Sender<()>,
 }
 
 //==================================================================================================
 // Implementations
 //==================================================================================================
 
 impl Vmm {
-    pub fn new(
+    ///
+    /// # Description
+    ///
+    /// This function instantiates and runs the virtual machine monitor (VMM) with the given arguments.
+    ///
+    /// # Parameters
+    ///
+    /// * `args` - Arguments for the virtual machine monitor.
+    ///
+    /// # Returns
+    ///
+    /// Upon successful completion, this method returns the exit status of the virtual machine.
+    /// Otherwise, it returns an error.
+    ///
+    pub fn spawn(
         memory_size: usize,
         kernel_filename: &str,
         initrd_filename: Option<String>,
         initrd_args: Option<String>,
         stderr: Option<String>,
         gateway_conn: Option<Gateway>,
-    ) -> Result<Self> {
+    ) -> Result<u16> {
         crate::timer!("vmm_creation");
 
         let (vm_tx, gateway_rx) = mpsc::channel::<Message>();
         let (gateway_tx, memory_thread_rx) = mpsc::channel::<Message>();
         let (memory_thread_tx, vm_rx) = mpsc::channel::<Message>();
+        let (snapshot_tx, snapshot_rx) = mpsc::channel::<SnapshotCommand>();
+        let (paused_tx, paused_rx) = mpsc::channel::<()>();
```

Comment:
These can be made more generic, and then we can it to more things, like telemetry:
- `let (snapshot_tx, snapshot_rx) = mpsc::channel::<SnapshotCommand>()` => `let (control_input_tx, control_input_rx) = mpsc::channel::<ControlCommand>()`
- `let (paused_tx, paused_rx) = mpsc::channel::<()>()` => `let (control_output_tx, control_output_rx) = mpsc::channel::<ControlCommandResponse>()`